core/src/test/java/io/grpc/internal/DnsNameResolverProviderTest.java

```diff
@@ -46,15 +51,52 @@ public void uncaughtException(Thread t, Throwable e) {
       .setServiceConfigParser(mock(ServiceConfigParser.class))
       .build();
 
-  private DnsNameResolverProvider provider = new DnsNameResolverProvider();
+  @Nullable
+  private String jndiSrvProperty;
+
+  @Before
+  public void memoizeSrvProperty() {
+    jndiSrvProperty = System.getProperty(ENABLE_GRPCLB_PROPERTY_NAME);
+  }
+
+  @After
+  public void restoreSrvProperty() {
+    if (jndiSrvProperty == null) {
+      System.clearProperty(ENABLE_GRPCLB_PROPERTY_NAME);
+    } else {
+      System.setProperty(ENABLE_GRPCLB_PROPERTY_NAME, jndiSrvProperty);
+    }
+  }
+
+  @Test
+  public void enableSrv_useDefault() {
+    System.clearProperty(ENABLE_GRPCLB_PROPERTY_NAME);
+
+    assertThat(new DnsNameResolverProvider().isSrvEnabled()).isFalse();
+  }
+
+  @Test
+  public void enableSrv_defaultEnabled() {
+    System.setProperty(ENABLE_GRPCLB_PROPERTY_NAME, "true");
+
+    assertThat(new DnsNameResolverProvider().isSrvEnabled()).isTrue();
+  }
+
+  @Test
+  public void enableSrv_defaultDisabled() {
+    System.setProperty(ENABLE_GRPCLB_PROPERTY_NAME, "false");
+
+    assertThat(new DnsNameResolverProvider().isSrvEnabled()).isFalse();
+  }
```

Comment:
Such test requires a behavior change (mutating the property at runtime would change the behavior of DnsNameResolvers created later, while it was not the case previously). Since the original DnsNameResolver didn't test with mutating the system property, you can probably leave it out too.